src/index.ts

```diff
@@ -36,8 +36,59 @@ export type NotarizeWaitOptions = NotarizeResult & NotarizeCredentials;
 export type NotarizeStapleOptions = Pick<NotarizeAppOptions, 'appPath'>;
 export type NotarizeOptions = NotarizeStartOptions;
 
-function authorizationArgs(opts: NotarizeCredentials): string[] {
-  if ('appleId' in opts) {
+function isPasswordCredentials(opts: NotarizeCredentials): opts is NotarizePasswordCredentials {
```

Comment:
> `(...): opts is NotarizePasswordCredentials`
I have never seen that syntax before, can you link me to the handbook reference?

Reply:
Sure thing: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards
From what I can tell, it provides a type hint to TypeScript for if statements using the condition. Seems it's typically used with [Union types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types) in the examples.